extensions/amp-experiment/1.0/attribute-allow-list/allowed-attribute-mutation-entry.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../../src/url';
+
+/**
+ * Base class for allowed attribute assertions.
+ * This should be extended to ensure the service
+ * is properly handled
+ */
+export class AllowedAttributeMutationEntry {
+  /**
+   * Constructor for the class
+   * @param {Array<string>=} opt_tags
+   */
+  constructor(opt_tags) {
+    this.tags = opt_tags;
+  }
+
+  /**
+   * Function to validate the value
+   * for the attribute change. Subclasses
+   * may override.
+   * @param {string} value
+   * @return {boolean}
+   */
+  validate(value) {
+    if (value) {
+      return true;
+    }
+  }
+
+  /**
+   * Function to apply the mutation.
+   * Subclasses may override
+   * @param {!Object} mutationRecord
+   */
+  mutate(mutationRecord) {
+    mutationRecord['targetElement'].setAttribute(
```

Comment:
I couldn't find it, but could you point me to where `mutationRecord['targetElement']` is defined? Thank you

Reply:
So where you commneted: "If the first argument is always going to be 'target', can we get rid of it?"

`setSelectorToElement()` is what sets the `targetElement` 😄